src/spyglass/ripple/v1/ripple.py

```diff
@@ -366,3 +361,128 @@ def plot_ripple(
         )
         ax.set_ylabel("LFPs")
         ax.set_xlabel("Time [s]")
+
+    def create_figurl(
+        self,
+        zscore_ripple=False,
+        ripple_times_color="red",
+        consensus_color="black",
+        speed_color="black",
+        view_height=800,
+        use_ripple_filtered_lfps=False,
+        lfp_offset=1,
+        lfp_channel_ind=None,
+    ):
+
+        ripple_times = self.fetch1_dataframe()
+        key = self.fetch1("KEY")
+        (
+            speed,
+            ripple_filtered_lfps,
+            sampling_frequency,
+        ) = self.get_ripple_lfps_and_position_info(key)
+        ripple_consensus_trace = self.get_Kay_ripple_consensus_trace(
+            ripple_filtered_lfps, sampling_frequency
+        )
+
+        if zscore_ripple:
+            ripple_consensus_trace = zscore(ripple_consensus_trace)
+
+        consensus_view = vv.TimeseriesGraph()
+        _add_ripple_times(consensus_view, ripple_times, ripple_times_color)
+        consensus_name = (
+            "Z-Scored Consensus Trace" if zscore_ripple else "Consensus Trace"
+        )
+        consensus_view.add_line_series(
+            name=consensus_name,
+            t=np.asarray(ripple_consensus_trace.index).squeeze(),
+            y=np.asarray(ripple_consensus_trace, dtype=np.float32).squeeze(),
+            color=consensus_color,
+            width=1,
+        )
+        if zscore_ripple:
+            ripple_params = (
+                RippleParameters
+                & {"ripple_param_name": key["ripple_param_name"]}
+            ).fetch1("ripple_param_dict")
+
+            zscore_threshold = ripple_params["ripple_detection_params"].get(
+                "zscore_threshold"
+            )
+            consensus_view.add_line_series(
+                name="Z-Score Threshold",
+                t=np.asarray(ripple_consensus_trace.index).squeeze(),
+                y=np.ones_like(
+                    ripple_consensus_trace, dtype=np.float32
+                ).squeeze()
+                * zscore_threshold,
+                color=ripple_times_color,
+                width=1,
+            )
+
+        if use_ripple_filtered_lfps:
+            interval_ripple_lfps = ripple_filtered_lfps
+        else:
+            lfp_merge_id = (LFPBandSelection & key).fetch1("lfp_merge_id")
+            lfp_df = (LFPOutput & {"merge_id": lfp_merge_id}).fetch1_dataframe()
+            interval_ripple_lfps = lfp_df.loc[speed.index[0] : speed.index[-1]]
+        if lfp_channel_ind is not None:
+            if lfp_channel_ind.max() >= interval_ripple_lfps.shape[1]:
+                raise ValueError(
+                    "lfp_channel_ind is out of range for the number of LFPs"
+                )
+            interval_ripple_lfps = interval_ripple_lfps.iloc[:, lfp_channel_ind]
+
+        lfp_view = vv.TimeseriesGraph()
```

Comment:
The use of multiple views here makes me think there should be a private method for each, but that's not necessary. Would someone ever want only `lfp_view`?

Reply:
Potentially earlier in the pipeline. Might be slightly different if you wanted to for example examine how good the filtering was. Could be a future improvement.